tools/ci/azure/tox_pytest.yml

```diff
@@ -0,0 +1,18 @@
+parameters:
+  directory: ''
+  toxenv: 'ALL'
+
+steps:
+- template: pip_install.yml
+  parameters:
+    packages: tox virtualenv
+
+- script: tox -c ${{ parameters.directory }} -e ${{ parameters.toxenv }} -- --junitxml=results.xml
+  displayName: 'Run tests'
+
+- task: PublishTestResults@2
```

Comment:
Out of curiosity (and I have to admit I haven't read the docs): is the `@2` a magic suffix?

Reply:
It's just the version of the tool, presumably so they can change it without existing content breaking.